application/gqlgen/message_test.go

```diff
@@ -0,0 +1,80 @@
+package gqlgen
+
+type messageResponse struct {
+	Message struct {
+		ID      string `json:"id"`
+		Content string `json:"content"`
+		Sender  struct {
+			ID       string `json:"id"`
+			Email    string `json:"email"`
+			Nickname string `json:"nickname"`
+			Location struct {
+				Description string `json:"description"`
+			} `json:"location"`
+		} `json:"sender"`
+		Thread struct {
+			ID           string `json:"id"`
+			Participants []struct {
+				ID       string `json:"id"`
+				Nickname string `json:"nickname"`
+			} `json:"participants"`
+		} `json:"thread"`
+	} `json:"message"`
+}
+
+func (gs *GqlgenSuite) TestMessageQuery() {
+	f := createFixtures_MessageQuery(gs)
+	c := getGqlClient()
+
+	query := `{ message(id: "` + f.Messages[0].Uuid.String() + `")
+		{ id content sender { location { description }} thread {id participants {nickname}}}}`
+
+	var resp messageResponse
+
+	TestUser = f.Users[0]
+	err := c.Post(query, &resp)
+	gs.NoError(err)
+
+	thread, err := f.Messages[0].GetThread([]string{"uuid", "id"})
+	gs.NoError(err)
+
+	participants, err := thread.GetParticipants([]string{"nickname"})
```

Comment:
What is the reason to test second- and third-level properties? Seems they will be tested in other places.

Reply:
Basically, I don't trust that just because they give good results as top level requests, they will give good results as nested requests (based on experience).